### PR TITLE
Refactor/error component

### PR DESCRIPTION
**Pull request by aliroemhildt**
## Checklist
- [ ] Tests written where required
- [ ] All tests passing
- [ ] Unused code/prys/etc deleted
Testing Coverage at <percentage>%

## Changes Made
  
What (if any) features are you implementing?
- n/a

What (if anything) did you refactor?
- pass error message to Error component in * route in App.js
- render UploadingSpinner when graphQL request is loading in AdminPage.js, SubmissionDetailsPage.js, LandingPage.js

Were there any issues that arose?
- n/a

Is there anything that you need from your teammate?
- review + merge

Any other comments, questions, or concerns?
- n/a

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ function App() {
           <Route exact path='/submissions' element={<AdminPage user={user}/>} />
           <Route exact path='/form' element={<SubmissionForm />} />
           <Route path='/submissions/:id' element={<SubmissionDetailsPage user={user}/>} />
-          <Route path='/*' element={<Error />} />
+          <Route path='/*' element={<Error error='incorrect url'/>} />
```
Comment (mr-ryan12) [inline src/App.js:24]: Thank you :)

File: src/components/AdminPage/AdminPage.js [hunk 2]
```diff
@@ -12,7 +13,13 @@ const AdminPage = ({ user }) => {
...
   }, [])
 
-  if (loading) return 'Loading...';
+  if (loading) {
+    return (
+      <div className="loading-container">
+        <UploadingSpinner />
+      </div>
+    );
+  }
```
Comment (mr-ryan12) [inline src/components/AdminPage/AdminPage.js:22]: Glad to see the animation put to good use! Excellent idea using it here!

File: src/components/SubmissionDetailsPage/SubmissionDetailsPage.js [hunk 3]
```diff
@@ -27,7 +28,13 @@ const SubmissionDetailsPage = ({ user }) => {
   const modal = showModal &&
     <Modal name={data.getSubmission.name} setShowModal={setShowModal} id={id}/>
   
-  if (loading) return <p>loading...</p>
```
Comment (mr-ryan12) [inline src/components/SubmissionDetailsPage/SubmissionDetailsPage.js:30]: Nice catch!

**Review by mr-ryan12 — APPROVED** [review]
Really appreciate all of the hard work you put into this. Thank you Ali!

**Review by Echory — APPROVED** [review]

**Review by oacortez — APPROVED** [review]
Thank you, Ali, for fixing these adjustments! I see you added our spinner animation for when data is loading to display. Very nice touch to let our users know! Thank you once again!

**Pull request merged**